app/src/main/java/iband/com/ibandplayerdemo/adapter/VariantAdapter.java

```diff
@@ -0,0 +1,65 @@
+package iband.com.ibandplayerdemo.adapter;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.support.annotation.NonNull;
+import android.support.v4.content.ContextCompat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+import java.util.ArrayList;
+import iband.com.ibandplayerdemo.R;
+
+/**
+ * Created by yossibarel on 18/07/2017.
+ */
+
+public class VariantAdapter extends ArrayAdapter<VariantHolder> {
+
+    private int mSelected;
+    private int mActive;
+    public VariantAdapter(@NonNull Context context, ArrayList<VariantHolder> variants) {
+        super(context, R.layout.item_variant, variants);
+    }
+
+    private static class ViewHolder {
+        TextView tvVariant;
+        View vCurrentVariant;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        VariantHolder variant = getItem(position);
+        ViewHolder viewHolder;
+        if (convertView == null) {
+            viewHolder = new ViewHolder();
```

Comment:
Move those commands to an informative function such as `setupView()`

Reply:
done

Reply:
in this scope i will need to init the convertView and the view holder ... so i cannot to separate this block